WW3:  Add indicator log file for gridded output

MatthewMasarik-NOAA opened the pull request:
# Description
Adds the functionality to write an indicator log file which will signal gridded output has completed writing for a given grid and timestep (Issue #3094)

Resolves #3094
Refs ufs-community/ufs-weather-model#2545
Refs NOAA-EMC/WW3#1321

Reviewer(s):  @KateFriedman-NOAA 

# Type of change
- [x] Bug fix (fixes something broken)
- [ ] New feature (adds functionality)
- [ ] Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? YES (a new file log file is written)
- Does this change require a documentation update? YES
- Does this change require an update to any of the following submodules? YES (WW3 within UFS-weather-model is updated)
  - [ ] EMC verif-global <!-- NOAA-EMC/EMC_verif-global#1234 -->
  - [ ] GDAS <!-- NOAA-EMC/GDASApp#1234 -->
  - [ ] GFS-utils <!-- NOAA-EMC/gfs-utils#1234 -->
  - [ ] GSI <!-- NOAA-EMC/GSI#1234 -->
  - [ ] GSI-monitor <!-- NOAA-EMC/GSI-Monitor#1234 -->
  - [ ] GSI-utils <!-- NOAA-EMC/GSI-Utils#1234 -->
  - [ ] UFS-utils <!-- ufs-community/UFS_UTILS#1234 -->
  - [x] UFS-weather-model NOAA-EMC/WW3#1347
  - [ ] wxflow <!-- NOAA-EMC/wxflow#1234 -->

# How has this been tested?
Clone, build, and run forecast-only on WCOSS2.

# Checklist
- [ ] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have documented my code, including function, input, and output descriptions
- [ ] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] This change is covered by an existing CI test or a new one has been added
- [x] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [ ] I have made corresponding changes to the system documentation if necessary

MatthewMasarik-NOAA committed:
Issue #3094: compound test conditions need double brackets, [[ ]]

MatthewMasarik-NOAA committed:
Issue #3094: fix loop logic bug involving "iwait"

MatthewMasarik-NOAA committed:
Issue #3094: add log file assignment

MatthewMasarik-NOAA committed:
Issue #3094: use $gfile in gfile_log definition, test cond, attribution

MatthewMasarik-NOAA committed:
Issue #3094: add log file linking to exglobal_forecast.sh

MatthewMasarik-NOAA committed:
Issue #3094: point to GFS.v16.3.1 -based ufs-weather-model in checkout.sh

MatthewMasarik-NOAA committed:
Issue #3094: correct UFS-WM branch in checkout.sh to wave-post/production/GFS.v16

KateFriedman-NOAA commented on sorc/checkout.sh:
```diff
@@ -23,7 +23,7 @@ echo $topdir
 if [[ ! -d fv3gfs.fd ]] ; then
     rm -f ${topdir}/checkout-fv3gfs.log
-    git clone --recursive --branch GFS.v16.3.1 https://github.com/ufs-community/ufs-weather-model.git fv3gfs.fd >> ${topdir}/checkout-fv3gfs.log 2>&1
+    git clone --recursive --branch wave-post/production/GFS.v16 https://github.com/MatthewMasarik-NOAA/ufs-weather-model.git fv3gfs.fd >> ${topdir}/checkout-fv3gfs.log 2>&1
```
Will complete review and approve once this is updated to be a new ufs-weather-model tag. Leaving comment here as reminder to self.

Also, when tag is ready, please update the `Externals.cfg` file at the top level of the branch with the new ufs-weather-model tag name. Thanks!

MatthewMasarik-NOAA replied:
Thank you @KateFriedman-NOAA, will do!

KateFriedman-NOAA reviewed (changes requested):
Changes to use new `.FINISHED` log file look good. I appreciate the extra space cleanup as well. Leaving a comment to change the ufs-weather-model tag name once those PRs are done and a new tag is cut. Please re-request a review from me once the tag is updated in both `sorc/checkout.sh` and `Externals.cfg`. Thanks!

KateFriedman-NOAA commented:
FYI @ADCollard, we will be merging a WW3 update into the GFSv16.4 release branch with this PR.

JessicaMeixner-NOAA commented:
@KateFriedman-NOAA - Are there other model changes in 16.4 or is this it?

KateFriedman-NOAA commented:
> @KateFriedman-NOAA - Are there other model changes in 16.4 or is this it?

This is it so far in terms of ufs-weather-model updates for v16.4. GFSv16.4 was originally a DA-only upgrade to bring `develop` GSI into production to help reduce work for GFSv17. See related issues #2987 and #3045. This will be a multi-component update now.

JessicaMeixner-NOAA commented:
Thanks @KateFriedman-NOAA  - Just wanted to make sure before we started asking for tags, etc.

MatthewMasarik-NOAA committed:
Issue #3094: update Externals.cfg for FV3GFS tag GFS.v16.4.0

MatthewMasarik-NOAA committed:
Issue #3094: update checkout.sh with ufs-community repo and tag GFS.v16.4.0

JessicaMeixner-NOAA commented:
@KateFriedman-NOAA The tag has been created and added to this PR.  Do you want any release notes updates? We can always add info in a second PR if needed to make things simpler.

KateFriedman-NOAA commented:
Thanks for the update @JessicaMeixner-NOAA ! Regarding the release notes, I will PR updates for that into @MatthewMasarik-NOAA 's PR branch. Stay tuned...

KateFriedman-NOAA commented:
Opened a PR into @MatthewMasarik-NOAA 's PR branch to add the initial release notes: https://github.com/MatthewMasarik-NOAA/global-workflow/pull/1

@MatthewMasarik-NOAA @JessicaMeixner-NOAA after you merge in my PR, please scan the updated release notes, add your name(s) to the bottom of it, and update any WW3-related details as needed. Thanks!

KateFriedman-NOAA committed:
Update release notes for WW3 changes (#1)

- create initial skeleton for GFSv16.4 release notes
- add WW3 script and ufs-weather-model tag updates

Refs #3094

MatthewMasarik-NOAA commented:
> add your name(s) to the bottom of it, and update any WW3-related details as needed.


> please scan the updated release notes, add your name(s) to the bottom of it, and update any WW3-related details as needed.

@KateFriedman-NOAA I was curious if the current description you have for the WW3 related changes is the level of detail that's wanted here, or should any more description be added?

MatthewMasarik-NOAA commented:
> > add your name(s) to the bottom of it, and update any WW3-related details as needed.
> 
> > please scan the updated release notes, add your name(s) to the bottom of it, and update any WW3-related details as needed.
> 
> @KateFriedman-NOAA I was curious if the current description you have for the WW3 related changes is the level of detail that's wanted here, or should any more description be added?

I spoke with @JessicaMeixner-NOAA offline about this and have an update I will push but up here momentarily.

MatthewMasarik-NOAA committed:
Issue #3094: Release_Notes.md - add some documentation for v16.4.0

MatthewMasarik-NOAA commented:
I added a short snippet for the WW3 related changes and our names.  @KateFriedman-NOAA @JessicaMeixner-NOAA please check though in case any edits are needed. Thank you

KateFriedman-NOAA reviewed (approved):
Looks good, thanks for the extra WW3 details in the release notes @MatthewMasarik-NOAA !

MatthewMasarik-NOAA commented:
> Looks good, thanks for the extra WW3 details in the release notes @MatthewMasarik-NOAA !

Sure thing!  Thank you @KateFriedman-NOAA.
